client/sites-dashboard-v2/index.tsx

```diff
@@ -103,7 +104,10 @@ const SitesDashboardV2 = ( {
 	const filteredSites = useSitesListFiltering( currentStatusGroup, {
 		search: dataViewsState.search,
 	} );
-	// todo: Perform pagination and sorting actions
+
+	// todo: Perform sorting actions
+
+	const paginatedSites = filteredSites.slice( ( page - 1 ) * perPage, page * perPage );
```

Comment:
We should probably use the dataViewsState page and perPage values, as the values from the query params coming from props are essentially used for initializing the dataViewState currently.

Since you are setting page on the query param based on the dataViewsState, it is getting updated here as expected (note we arent doing that with perPage yet so those settings don't yet work). This approach should work in general, but so far we have only been using the query params for initialization and dataViewsStates for usage. However, the pagination seems to get reset on mount (if they initialize with page=2, for some reason they instantly go back to setting page=1), to me there doesn't seem like much of a benefit to keeping track of them in the URL while this is happening anyways.